libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/service/NavigationNotificationService.kt

```diff
@@ -46,4 +52,44 @@ internal class NavigationNotificationService : Service() {
         stopForeground(true)
         MapboxTripService.unregisterOneTimeNotificationDataObserver(notificationDataObserver)
     }
+
+    @CallSuper
+    override fun dump(
+        fd: FileDescriptor, writer: PrintWriter,
+        args: Array<String>?
+    ) {
+        super.dump(fd, writer, args)
+        for (arg in args!!) {
+            when (arg) {
+                DUMPSYS_ARG_UNIT_TYPE_IMPERIAL -> updateDistanceFormatUnitType(UnitType.IMPERIAL)
```

Comment:
1) Why can we only change the unit type?
2) In the usage example you provided I see that only unit type itself is passed as an argument. Shouldn't we specify somehow that it's a unit type? It would be even more relevant if we add the ability to change something else. For example, we could require the command to be sth like this:
```
adb shell dumpsys activity service com.mapbox.navigation.core.trip.service.NavigationNotificationService unit_type=UNIT_TYPE_IMPERIAL
```
And then split the arguments by "=".

Reply:
Yeah the arg comes in as a full string "unit_type=UNIT_TYPE_IMPERIAL".

Considering it requires another system to parse a format (it needs to handle spaces? unit_type = UNIT_TYPE_IMPERIAL).

Went for the quick and easy format. Considering how we may want to add the ability to do more advanced things like, request route. It may be best to define a JSON format for all our potential arguments.

If we define something like `{"type": "<whatever>", "json": "<json-string>"}`

Then we could define anything we wanted
````
{
    "type": "set_directions_route",
    "json": "{"country_crossed":false,"weight_typical":272.769,"routeIndex":"0",..,
}
````

Reply:
Well json isn't really supported, i went deep into this. 

But this now allows you to pass arguments :)